packages/wallet-connect-sdk-core/src/index.ts

```diff
@@ -262,33 +607,106 @@ export class WcSdk {
+     *
+     * const from = { type: 'Address', value: senderAddress }
+     * const recipient = { type: 'Address', value: 'NbnjKGMBJzJ6j5PHeYhjJDaQ5Vy5UYu4Fv' }
+     * const value = { type: 'Integer', value: 100000000 }
+     * const args = { type: 'Array', value: [] }
+     *
+     * const resp = await WcSdk.invokeFunction(wcClient, session, chainId, {
+     *    scriptHash: smartContractScripthash,
+     *    operation: 'transfer',
+     *    args: [from, recipient, value, args]
+     * })
+     * ```
+     * @param wcClient
+     * @param session connected session
+     * @param chainId the chosen blockchain id to make the request, must be one of the blockchains authorized by the wallet
+     * @param request the contract invocation options
+     * @return the call result promise. It might only contain the transactionId, another call to the blockchain might be necessary to check the result.
+     */
+    static async invokeFunction (wcClient: Client, session: SessionTypes.Created, chainId: string, request: ContractInvocation): Promise<RpcCallResult> {
         return WcSdk.sendRequest(wcClient, session, chainId, {
             method: 'invokefunction',
             params: [request],
         })
     }
 
-    static async testInvoke(wcClient: Client, session: SessionTypes.Created, chainId: string, request: ContractInvocation) {
+    /**
+     * Sends a `testInvoke` request to the Wallet and it will communicate with the blockchain.
+     * It will not consume any gas but it will also not persist any data, this is often used to retrieve SmartContract information or check how much gas an invocation will cost.
+     * Also, the wallet might choose to not ask the user authorization for test invocations making them easy to use.
+     * ```
+     * const resp = await WcSdk.testInvoke(wcClient, session, chainId, {
+     *     scriptHash: '0x010101c0775af568185025b0ce43cfaa9b990a2a',
+     *     operation: 'getStream',
+     *     args: [{ type: 'Integer', value: 17 }],
+     *     signer: { scopes: WitnessScope.None }
+     * })
+     * ```
+     * @param wcClient
+     * @param session connected session
+     * @param chainId the chosen blockchain id to make the request, must be one of the blockchains authorized by the wallet
+     * @param request the contract invocation options
+     * @return the call result promise
+     */
+    static async testInvoke (wcClient: Client, session: SessionTypes.Created, chainId: string, request: ContractInvocation): Promise<RpcCallResult> {
         return WcSdk.sendRequest(wcClient, session, chainId, {
             method: 'testInvoke',
             params: [request],
         })
     }
 
-    static async multiInvoke(wcClient: Client, session: SessionTypes.Created, chainId: string, request: ContractInvocationMulti) {
+    /**
+     * Sends a `multiInvoke` request to the Wallet, it will concatenate all invocations and send to the blockchain all at once.

```

Comment:
```suggestion
     * Sends a `multiInvoke` request to the Wallet. It will concatenate all invocations and send them to the blockchain all at once.

```